datadome/resource_custom_rule.go

```diff
@@ -120,23 +121,23 @@ func resourceCustomRuleCreate(ctx context.Context, d *schema.ResourceData, m int
 		Enabled:      d.Get("enabled").(bool),
 	}
 
-	o, err := c.CreateCustomRule(ctx, newCustomRule)
+	id, err := c.Create(ctx, newCustomRule)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	d.SetId(strconv.Itoa(o.ID))
+	d.SetId(strconv.Itoa(*id))
```

Comment:
There was a little update between the old `CreateCustomRule` and the `Create` method from this PR.

In the `CreateCustomRule` method, on [this line](https://github.com/DataDome/terraform-provider-datadome/pull/17/files#diff-19eae9f8aef7b444dae5398c59b7521c1b4e9cae48c2d216229b7deb5850f04dR64), we were returning the `id` which is a structure defining an `ID` field in it (source code [right here](https://github.com/DataDome/terraform-provider-datadome/blob/051ba5a170fcfe221a697ab6dcff6373f0becf5b/datadome-client-go/models.go#L23)).

In the `Create` method, instead of passing an `ID` struct we are giving the address of `id.ID` (this part [of code](https://github.com/DataDome/terraform-provider-datadome/pull/25/files#diff-0b123e04e54112fba6aa7300e03117a625df08b3fcb847f56d83c39efd7c365bR137)).

For the `Update` method, we are returning the whole `CustomRule` including the ID field in it. That's why we have those differences between these methods.